File: app/src/main/java/es/etologic/mahjongscoring2/domain/use_cases/GetPlayerUseCase.java
package es.etologic.mahjongscoring2.domain.use_cases;

import javax.inject.Inject;

import es.etologic.mahjongscoring2.data.repositories.PlayersRepository;
import es.etologic.mahjongscoring2.domain.entities.Player;
import io.reactivex.Single;

public class GetPlayerUseCase {

    private final PlayersRepository playersRepository;

    @Inject
    public GetPlayerUseCase(PlayersRepository playersRepository) { this.playersRepository = playersRepository; }

    public Single<Player> getPlayer(Long playerId) {
        return playersRepository.getOne(playerId);
    }
}

File: app/src/main/java/es/etologic/mahjongscoring2/domain/use_cases/GetPlayerUseCase.kt
package es.etologic.mahjongscoring2.domain.use_cases

import es.etologic.mahjongscoring2.data.repositories.PlayersRepository
import es.etologic.mahjongscoring2.domain.model.Player
import io.reactivex.Single
import javax.inject.Inject

class GetPlayerUseCase @Inject
constructor(private val playersRepository: PlayersRepository) {
    
    fun getPlayer(playerId: Long): Single<Player> {
        return playersRepository.getOne(playerId)
    }
}
